src/Uno.UI/UI/Xaml/Controls/ComboBox/ComboBox.cs

```diff
@@ -232,6 +232,64 @@ private void OnHeaderTemplateChanged(DataTemplate oldHeaderTemplate, DataTemplat
 			UpdateHeaderVisibility();
 		}
 
+		protected override void PrepareContainerForItemOverride(DependencyObject element, object item)
+		{
+			SynchronizeContainerInheritedProperties(element, item);
+
+			base.PrepareContainerForItemOverride(element, item);
+		}
+
+		protected override void ClearContainerForItemOverride(DependencyObject element, object item)
+		{
+			ClearContainerInheritedProperties(element, item);
+
+			base.ClearContainerForItemOverride(element, item);
+		}
+
+		/// <summary>
+		/// Explicitly propagate inherited properties values to the item containers, as they are not direct visual children of the ComboBox.
+		/// </summary>
+		private void SynchronizeContainerInheritedProperties(DependencyObject element, object item)
+		{
+			if (element is ContentControl cc && !IsItemItsOwnContainer(item))
```

Comment:
Is the `element is ContentControl cc` check needed?

Do we have a test case where `element` is not a `ContentControl`?

Reply:
If it's not a ContentControl (so not an `ComboBoxItem`), `IsItemItsOwnContainer` will be false. We already have test cases for non-ComboBoxItem items.

Reply:
@jeromelaban Doesn't that mean `element is ContentControl` is unnecessary check (ie, `!IsItemItsOwnContainer(item)` is sufficient alone)?